tests/functional/aws-node-sdk/test/bucket/putBucketTagging.js

```diff
@@ -0,0 +1,177 @@
+    TagSet: [
+        {
+            Key: 'string',
+            Value: 'string',
+        },
+        {
+            Key: 'string',
+            Value: 'stringaaaa',
+        },
+    ],
+};
+
+const validTagging = {
+    TagSet: [
+        {
+            Key: 'key1',
+            Value: 'value1',
+        },
+        {
+            Key: 'key2',
+            Value: 'value2',
+        },
+    ],
+};
+
+const validSingleTagging = {
+    TagSet: [
+        {
+            Key: 'key1',
+            Value: 'value1',
+        },
+    ],
+};
+
+const validEmptyTagging = {
+    TagSet: [],
+};
+
+const taggingKeyNotValid = {
+    TagSet: [
+        {
+            Key: 'stringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaastringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' +
+                'astringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaastringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' +
+                'stringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+            Value: 'string',
+        },
+        {
+            Key: 'string',
+            Value: 'stringaaaa',
+        },
+    ],
+};
+
+const taggingValueNotValid = {
+    TagSet: [
+        {
+            Key: 'stringaaa',
+            Value: 'string',
+        },
+        {
+            Key: 'string',
+            Value: 'stringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaastringaaaaaaaaaaaaaaaaaaaaaa' +
+                'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaastringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' +
+                'aaaaaaastringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaastringaaaaaaaaaaaaaaaaaa' +
+                'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaastringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' +
+                'aaaaaaaaaaastringaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaastringaaaaaaaaaaaaaa' +
+                'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+        },
+    ],
+};
+
+// Check for the expected error response code and status code.
+function assertError(err, expectedErr, cb) {
+    if (expectedErr === null) {
+        assert.strictEqual(err, null, `expected no error but got '${err}'`);
+    } else {
+        assert.strictEqual(err.code, expectedErr, 'incorrect error response ' +
+            `code: should be '${expectedErr}' but got '${err.code}'`);
+        assert.strictEqual(err.statusCode, errors[expectedErr].code,
+            'incorrect error status code: should be  ' +
+            `${errors[expectedErr].code}, but got '${err.statusCode}'`);
+    }
+    cb();
+}
+
+describe('aws-sdk test put bucket tagging', () => {
+    let s3;
+
+    before(done => {
+        const config = getConfig('default', { signatureVersion: 'v4' });
+        s3 = new S3(config);
+        return done();
+    });
+
+    beforeEach(done => s3.createBucket({ Bucket: bucket }, done));
+
+    afterEach(done => s3.deleteBucket({ Bucket: bucket }, done));
+
+    it('should not add tag if tagKey not unique', done => {
```

Comment:
do you think it's worth testing if a bucket tagging can overwrite the one with the same key?

Reply:
It does this by default, all tags are overwritten every time a tagset is uploaded